src/vsl/parser/parser.ne

```diff
@@ -0,0 +1,430 @@
+# VSL - Primary Parser
+@{%
+let t = require('./nodes');
+let lexer = new (require('./vsltokenizer'))();
+
+const NodeTypes = require('./vsltokentype'),
+  freeze = Object.freeze,
+  integer = freeze({ test: x => x[1] === NodeTypes.Integer }),
+  decimal = freeze({ test: x => x[1] === NodeTypes.Decimal }),
+  string = freeze({ test: x => x[1] === NodeTypes.String }),
+  identifier = freeze({ test: x => x[1] === NodeTypes.Identifier }),
+  special_loop = freeze({ test: x => x[1] === NodeTypes.SpecialArgument }),
+  special_identifier = freeze({ test: x => x[1] ===
+            NodeTypes.SpecialIdentifier }),
+  unwrap = d => d[0].value,
+  mid = d => d[0][0];
+%}
+
+@lexer lexer
+@split false
+@has false
+
+@builtin "postprocessors.ne"
+
+CodeBlock[s]
```

Comment:
This is kinda difficult to read being so long so I think adding some:

```ruby
# ================== Data Structures ================
```

would help give a sense of hierarchy in the parser. Though @somebody1234 has worked on parser most and I don't want this to be non-compatible with his branch >_<